### PR TITLE
Refactored migration of rbac resource rules

### PR DESCRIPTION
**Pull request by vicpatel**
Please take a look.

### CODE REVIEW

File: pkg/provision/migrations_202203_rbac_resource_fix.go [hunk 1]
```diff
@@ -118,149 +206,170 @@ func migratePost202203RbacRule(r *rbac.Rule, rx *resIndex) (op int) {
...
+func preloadRbacResourceIndex(ctx context.Context, s store.Storer, res fetchResIndex) (*resIndex, error) {
 	rx := &resIndex{}
 
 	rx.records = make(map[uint64]*composeTypes.Record)
 	var getRecords func(*composeTypes.Module, *filter.PagingCursor) (composeTypes.RecordSet, composeTypes.RecordFilter, error)
 	getRecords = func(m *composeTypes.Module, cursor *filter.PagingCursor) (rr composeTypes.RecordSet, f composeTypes.RecordFilter, err error) {
 		rr, f, err = store.SearchComposeRecords(ctx, s, m, composeTypes.RecordFilter{
 			Paging: filter.Paging{
-				Limit:      10000,
+				Limit:      30000,
```
Comment (tjerman) [inline pkg/provision/migrations_202203_rbac_resource_fix.go]: Why make this larger? Even the original 10k seems a bit extreme to me.

File: pkg/provision/migrations_202203_rbac_resource_fix.go [hunk 2]
```diff
@@ -118,149 +206,170 @@ func migratePost202203RbacRule(r *rbac.Rule, rx *resIndex) (op int) {
...
 
-			r.Resource = composeTypes.ModuleFieldRbacResource(p1, p2, ID)
+	partsLen := len(parts)
+	ID := uint64(0)
+	p1 := uint64(0)
+	p2 := uint64(0)
+	parseUint := func(s string) uint64 {
+		id, _ := strconv.ParseUint(s, 10, 64)
+		return id
+	}
```
Comment (tjerman) [inline pkg/provision/migrations_202203_rbac_resource_fix.go]: You can use the `cast.ToUint64()` (the one without the error) to get rid of this custom bit

**Review by tjerman — COMMENTED** [review]
To me, it seems fine; go through the few comments I've left and consider them.

The only requests I'd have is if you could write a nice long comment above `migratePost202203RbacRules` what exactly this does and why it exists, you replacing the `-1`, `0`, `1` *magic numbers* denoting operations with either a constant (so eg. `actionRemove`, `actionNone`, `actionUpdate`) or a new unexported type.

File: pkg/provision/migrations_202203_rbac_resource_fix.go (hunk 1 above)
Comment (vicpatel) [reply pkg/provision/migrations_202203_rbac_resource_fix.go]: True, I will make it 2k just in case.

**Pull request merged**